### PR TITLE
Expand smaps / smaps_rollup parsing

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit expands the number of fields we parse from the smaps/smaps_rollup
files. I have not hooked these up yet to metrics emission but will in a follow-up
commit to this PR.

### CODE REVIEW

**Comment by blt** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/DataDog/lading/1130?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1130** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1130?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/DataDog/lading/1130?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1129** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1129?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1128** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1128?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1127** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1127?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1124** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1124?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1120** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1120?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1125** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1125?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1126** <a href="https://app.graphite.dev/github/pr/DataDog/lading/1126?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lading/src/observer/linux/procfs/memory/smaps.rs [hunk 1]
```diff
@@ -184,6 +245,20 @@ impl Region {
...
+            referenced: referenced.unwrap_or(0),
+            anonymous: anonymous.unwrap_or(0),
+            lazy_free: lazy_free.unwrap_or(0),
+            anon_huge_pages: anon_huge_pages.unwrap_or(0),
+            shmem_pmd_mapped: shmem_pmd_mapped.unwrap_or(0),
+            file_pmd_mapped: file_pmd_mapped.unwrap_or(0),
+            shared_hugetlb: shared_hugetlb.unwrap_or(0),
+            private_hugetlb: private_hugetlb.unwrap_or(0),
+            swap_pss: swap_pss.unwrap_or(0),
+            locked: locked.unwrap_or(0),
```
Comment (GeorgeHahn) [inline lading/src/observer/linux/procfs/memory/smaps.rs:261]: Some preference towards doing the same thing with `size, pss, rss, and swap` to make handling consistent unless there's a reason to have different behavior for those.

**Review by GeorgeHahn — APPROVED** [review]

File: lading/src/observer/linux/procfs/memory/smaps.rs (hunk 1 above)
Comment (scottopell) [reply lading/src/observer/linux/procfs/memory/smaps.rs:261]: Agreed, I'd like to see the required fields unwrapped similarly to how `size, pss, rss` etc are unwrapped above. 
Only a few of these fields are actually optional, and I don't want to report `0` incorrectly.

**Review by scottopell — CHANGES_REQUESTED** [review]

File: lading/src/observer/linux/procfs/memory/smaps.rs (hunk 1 above)
Comment (blt) [reply lading/src/observer/linux/procfs/memory/smaps.rs:261]: Hmm the more I look at this the more I think the current approach is bankrupt. Best to read the whole file in, rip through it and create metrics without the intermediary structure.

Comment (scottopell) [reply lading/src/observer/linux/procfs/memory/smaps.rs]: new approach seems good in terms of simpler code, but I'm concerned that we're blindly reporting 0 for any values that don't exist. 
Unless I'm misunderstanding the code here, but it seems like we would end up reporting `0` in cases where there is no value, which makes this data confusing to work with.

**Review by scottopell — APPROVED** [review]

**Comment by blt** [comment]
### Merge activity

* **Dec 6, 2:23 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/DataDog/lading/1130).

**Pull request merged**